src/main/java/thaumicenergistics/common/inventory/HandlerEssentiaStorageBusContainer.java

```diff
@@ -577,12 +577,12 @@ public boolean validForPass( final int pass )
 			// Is this the priority pass?
 			if( pass == 1 )
 			{
-				// Valid if has filters or container has something in it
-				return( hasFilters || hasStored );
+				// Valid if has whitelist or container has something in it
+				return( ( hasFilters && !this.inverted ) || hasStored );
```

Comment:
iirc if there is a filter, be it blacklist or whitelist, it should be handled first

Reply:
I looked up vanilla ae2 storage bus when touching this. It also only prioritize when it has a whitelist. See appeng.me.storage.MEInventoryHandler#isPrioritized.